docs/DeveloperGuide.md

```diff
@@ -1,38 +1,461 @@
+For example, a `delete` command will cause `Parser` to create a new instance of `DeleteCommand(pid)`.
+6. The created command object is returned back to `main` function for further processing.
+
+### Commands component
+
+**API** : [`Command.java`](https://github.com/AY2324S2-CS2113T-T09-3/tp/blob/master/src/main/java/seedu/stockpal/commands/Command.java)
+
+
+![Structure of the Commands Component](images/CommandsClassDiagram.png)
+
+### Data component
+
+**API** : [`Data`](https://github.com/AY2324S2-CS2113T-T09-3/tp/blob/master/src/main/java/seedu/stockpal/data)
+
+The following is a class diagram of the `data` component.
+![Structure of the Data Component](images/DataClassDiagram.png)
+
+
+### Storage component
+
+**API** : [`Storage.java`](https://github.com/AY2324S2-CS2113T-T09-3/tp/blob/master/src/main/java/seedu/stockpal/storage/Storage.java)
+
+![Structure of the Storage Component](images/StorageClassDiagram.png)
+
+The `Storage` component,
+* can save product list data in CSV format, and load them back into corresponding Products.
+* depends on the `StockPal` component (because the `Storage` component's job is to save/load objects that belong to `StockPal`)
+* consists of the classes `Storage`, `CsvWriter` and `CsvReader`.
+  * `Storage` defines methods that loads and saves data.
+  * `CsvWriter` is responsible for handling the writing of data to the CSV data file.
+  * `CsvReader` is responsible for handling the reading of data from the CSV data file.
+
+### Common classes
+
+Classes used by multiple components are in the `seedu.stockpal.common` package.
+
+### Exception classes
+
+Exceptions classes used by multiple components are in the `seedu.stockpal.exceptions` package.
+
+--------------------------------------------------------------------------------------------------------------------
+
+## **Implementation**
+
+This section describes some noteworthy details on how certain features are implemented.
+
+### Add product feature
+
+The NewCommand class is responsible for adding a new product to the inventory in the StockPal application.
+
+#### Class diagram of NewCommand:
+<img src="images/AddCommandClass.png" alt=""/>
+
+
+**Attributes**
+* name: The name of the product.
+* quantity: The initial quantity of the product.
+* price: The price of the product.
+* description: The description of the product.
+
+**Methods**
+* `NewCommand`: Constructor for creating a new instance of the NewCommand class.
+* `execute`: Method to add the new product to the product list.
+* `createProduct`: Method to create a new product with a unique product ID.
+
+The following sequence diagram shows how an add operation works when the user inputs a valid `add` command:
+<img src="images/AddCommandSequence.png" alt="AddCommandSequence.png"/>
+
+The following sequence diagram details how `AddCommand#execute()` functions.
+<img src="images/AddCommandExecuteSequenceDiagram.png" alt="AddCommandExecuteSequenceDiagram.png"/>
```

Comment:
Sequence diagram is very thorough. One small thing to fix would be the colour of the alt conditions as they blend with the dark red bars, making it a little difficult to read. could look into using a different colour